tests/test_browser.py

```diff
@@ -5392,6 +5392,17 @@ def test_assert_failure(self):
   def test_full_js_library_strict(self):
     self.btest_exit(test_file('hello_world.c'), args=['-sINCLUDE_FULL_LIBRARY', '-sSTRICT_JS'])
 
+  def test_dynamic_link_with_prejs(self):
+    create_file('script.js', r'''
+      console.log('Foo');
+    ''')
+    create_file('main.c', r'''
+      int main( int argc, const char *argv[] ) {
+        return 0;
+      }
+    ''')
+    self.btest_exit('main.c', args=['-sMAIN_MODULE=2', '--pre-js=script.js'])
```

Comment:
Is there a non-ascii character here? From the test name I'm not sure if it should, but I'm also not sure how it relates to the utf-8 fix in this PR?

Reply:
Non-ascii symbol - ↓:
https://github.com/emscripten-core/emscripten/blob/2a3bb65cdea614451e4dc6227ae8e338902bfd1d/src/library_dylink.js#L534

My test hits this line:
https://github.com/emscripten-core/emscripten/blob/2a3bb65cdea614451e4dc6227ae8e338902bfd1d/emcc.py#L2856

And there is UnicodeEncodeError because of symbol above.

You can try to run my test without fix and you will see the problem.

Reply:
Wouldn't it make sense to puts some non-UTF characaters in the pre.js file that is part of the test?   (In case somebody deletes that TODO line).

Also, that does currently passes for me without your patch, so I'm guessing this must be some kind of OS-specific or environment specific issue?   What OS are you running on?  Are you running on a non-default locale?

Also, can you move this tests into test_other.py?   I don't think there is any need to run this in the browser.

Reply:
I'm curious, does `core2.test_unicode_js_library` fail for you locally (without your patch)?

Reply:
I'm on Windows, default locale I believe.

Reply:
`core2.test_unicode_js_library` passes without fix.

Reply:
Ah I guess because it does not contain a pre-js file... I think we should add a unicode `--pro-js` to that tests.

In the mean time I'm trying to ensure we catch these kinds of errors in CI: #17118